autoPyTorch/pipeline/components/base_component.py

```diff
@@ -252,3 +252,36 @@ def __str__(self) -> str:
         """Representation of the current Component"""
         name = self.get_properties()['name']
         return "autoPyTorch.pipeline %s" % name
+
+    def _apply_search_space_update(self, name: str, new_value_range: Union[List, Tuple],
+                                   default_value: Union[int, float, str], log: bool = False) -> None:
+        """Allows the user to update a hyperparameter
+
+        Arguments:
+            name {string} -- name of hyperparameter
+            new_value_range {List[?] -- value range can be either lower, upper or a list of possible conditionals
+            log {bool} -- is hyperparameter logscale
+        """
+
+        if len(new_value_range) == 0:
+            raise ValueError("The new value range needs at least one value")
+        self._cs_updates[name] = tuple([new_value_range, default_value, log])
+
+    def _get_search_space_updates(self, prefix: Optional[str] = None) -> Dict[str, Tuple]:
+        """Get the search space updates with the given prefix
+
+        Keyword Arguments:
+            prefix {str} -- Only return search space updates with given prefix (default: {None})
+
+        Returns:
+            dict -- Mapping of search space updates. Keys don't contain the prefix.
+        """
+        if prefix is None:
+            return self._cs_updates
+        result: Dict[str, Tuple] = dict()
+
+        # iterate over all search space updates of this node and filter the ones out, that have the given prefix
+        for key in self._cs_updates.keys():
+            if key.startswith(prefix):
+                result[key[len(prefix):]] = self._cs_updates[key]
```

Comment:
In the base choice, you added `len(prefix):+1` whereas here `len(prefix):`. Why is that so? Can you add a comment in the code for this?
Also, can you update the comment:
```
# iterate over all search space updates of this node and filter the ones out, that have the given prefix
```

Shouldn't it indicate that you are including the ones with the prefix not filtering them out?

Reply:
Its because component hyperparameters will not have a delimiter. I'll add this to a comment